code/game/objects/items/weapons/storage/boxes.dm

```diff
@@ -981,3 +981,304 @@
 	new /obj/item/weapon/reagent_containers/glass/beaker/erlenmeyer/sodawater(src)
 	new /obj/item/weapon/reagent_containers/glass/beaker/large/erlenmeyer(src)
 	new /obj/item/weapon/paper/diy_soda(src)
+
+
+//Smart boxes.
+/obj/item/weapon/storage/box/smartbox
+	name = "Smart-box"
+	desc = "A one-use box that leaves no trash or cardboard behind."
+	foldable = null
+	storage_slots = BOX_SPACE
+	var/one_way = 0 //For one way boxes, you can take out but not put in. Could be moved to /box.
+
+
+/obj/item/weapon/storage/box/smartbox/remove_from_storage(obj/item/W as obj, atom/new_location, var/force = 0, var/refresh = 1)
+	. = ..()
+	if(contents.len <= 0) //If this is the last item, kill the box.
+		to_chat(new_location,"<span class='notice'>The [src] fizzles away into a glittering dust.</span>") //Should probably typecheck new_location as a mob?
```

Comment:
`new_location` is not guaranteed to be a player mob, you can't call `to_chat(some_turf)`
Just make this a `visible_message()`